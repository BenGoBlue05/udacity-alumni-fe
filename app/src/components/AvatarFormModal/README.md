## AvatarFormModal Component
A component that ...

### Example

```js
<AvatarFormModal />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **myProp**    | String   |             | Any string value


### Other Information
