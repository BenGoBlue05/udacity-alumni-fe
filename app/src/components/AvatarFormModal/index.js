import React, { PropTypes } from 'react';
import { StyledWrapper } from './styles';
import Layer from 'grommet-udacity/components/Layer';
import Form from 'grommet-udacity/components/Form';
import FormField from 'grommet-udacity/components/FormField';
import Box from 'grommet-udacity/components/Box';
import Heading from 'grommet-udacity/components/Heading';
import Footer from 'grommet-udacity/components/Footer';
import Button from 'grommet-udacity/components/Button';
import Menu from 'grommet-udacity/components/Menu';
import CheckmarkIcon from 'grommet-udacity/components/icons/base/Checkmark';
import CloseIcon from 'grommet-udacity/components/icons/base/Close';
import { Avatar } from 'components';

const AvatarFormModal = ({
  isVisible,
  onClose,
  onSave,
  onCancel,
  onChange,
  avatarString,
  user,
}) => (
  <Layer
    closer
    onClose={onClose}
    hidden={!isVisible}
    align="center"
  >
    <Box pad="large">
      <Box pad="medium">
        <Heading align="center" tag="h2">
          {`Edit Avatar for ${user ? user.name : ''}`}
        </Heading>
      </Box>
      <Box align="center" justify="center" pad="medium">
        <Avatar src={user ? user.avatar : null} />
      </Box>
      <Form>
        <FormField
          label="Avatar"
          help="Enter a URL to set this User's Avatar"
        >
          <input
            value={avatarString}
            onChange={onChange}
            placeholder="https://c2.staticflickr.com/8/7127/7552248154_978bcb1773.jpg"
            type="text"
          />
        </FormField>
      </Form>
      <Footer align="center" justify="center">
        <Menu inline direction="row" responsive={false}>
          <Button
            label="Cancel"
            secondary
            style={{ marginTop: 10, marginRight: 5 }}
            onClick={onCancel}
            icon={<CloseIcon />}
          />
          <Button
            label="Save"
            primary
            style={{ marginTop: 10, marginLeft: 5 }}
            onClick={onSave}
            icon={<CheckmarkIcon />}
          />
        </Menu>
      </Footer>
    </Box>
  </Layer>
);

AvatarFormModal.propTypes = {
  isVisible: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  onSave: PropTypes.func.isRequired,
  onCancel: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  avatarString: PropTypes.string,
  user: PropTypes.object,
};

export default AvatarFormModal;
