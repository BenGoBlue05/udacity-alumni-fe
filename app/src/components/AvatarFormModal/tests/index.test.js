import AvatarFormModal from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<AvatarFormModal />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <AvatarFormModal />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
