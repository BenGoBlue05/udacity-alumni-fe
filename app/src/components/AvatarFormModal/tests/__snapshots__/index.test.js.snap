exports[`<AvatarFormModal /> should render with default props 1`] = `
<Layer
  align="center"
  closer={true}
  hidden={true}>
  <Box
    announce={false}
    direction="column"
    focusable={true}
    pad="large"
    responsive={true}
    tag="div">
    <Box
      align="center"
      announce={false}
      direction="column"
      focusable={true}
      justify="center"
      pad="medium"
      responsive={true}
      tag="div">
      <undefined
        src={null} />
      <Heading
        align="center"
        tag="h2" />
    </Box>
    <Form
      compact={false}
      fill={false}
      pad="none">
      <FormField
        help=""
        label="Avatar"
        size="medium"
        strong={false}>
        <input
          placeholder="https://c2.staticflickr.com/8/7127/7552248154_978bcb1773.jpg"
          type="text" />
      </FormField>
    </Form>
    <Footer
      align="center"
      direction="row"
      justify="center"
      responsive={false}>
      <Menu
        closeOnClick={true}
        direction="row"
        dropAlign={
          Object {
            "left": "left",
            "top": "top"
          }
        }
        inline={true}
        pad="none"
        responsive={false}>
        <Button
          icon={
            <Close
              a11yTitleId="close-title"
              responsive={true} />
          }
          label="Cancel"
          secondary={true}
          style={
            Object {
              "marginRight": 5,
              "marginTop": 10
            }
          }
          type="button" />
        <Button
          icon={
            <Checkmark
              a11yTitleId="checkmark-title"
              responsive={true} />
          }
          label="Save"
          primary={true}
          style={
            Object {
              "marginLeft": 5,
              "marginTop": 10
            }
          }
          type="button" />
      </Menu>
    </Footer>
  </Box>
</Layer>
`;
